Task1/filter/filter.js

```diff
@@ -1,15 +1,23 @@
-function filter(arr, condition){
-  let newArr = []
-  for(let i = 0; i < arr.length; i++){
-    if(condition(arr[i])){
-      newArr.push(arr[i])
+function filter(array, condition){
+  const newArr = []
+  if(typeof condition != "function"){
+    console.log("Error:'filter' function second argument was not a function")
+  }
```

Comment:
В будущем было бы здорово использовать следующее оформление структуры if.. else if:
```suggestion
  } else if(condition.length !== 1) {
    //some code
  } else {
   //some code
  }
```